net/dns/wsl_windows.go

```diff
@@ -225,7 +204,34 @@ func wslCombinedOutput(cmd *exec.Cmd) ([]byte, error) {
 	cmd.Stdout = buf
 	cmd.Stderr = buf
 	err := wslRun(cmd)
-	return buf.Bytes(), err
+
+	// Some of wsl.exe's output get printed as UTF-16, which breaks a
+	// bunch of things. Try to detect this by looking for a zero byte
+	// in the first few bytes of output (which will appear if any of
+	// those codepoints are basic ASCII - very likely). From that we
+	// can infer that UTF-16 is being printed, and the byte order in
+	// use, and we decode that back to UTF-8.
+	//
+	// https://github.com/microsoft/WSL/issues/4607
+	ret := buf.Bytes()
+	checkLen := 20
+	if len(ret) < checkLen {
+		checkLen = len(ret)
+	}
+	if zeroOff := bytes.IndexByte(ret[:checkLen], 0); zeroOff != -1 {
+		// We assume wsl.exe is trying to print an ascii codepoint,
+		// meaning the zero byte is in the upper 8 bits of the
+		// codepoint. That means we can use the zero's byte offset to
+		// work out if we're seeing little-endian or big-endian
+		// UTF-16.
+		var endian binary.ByteOrder = binary.LittleEndian
```

Comment:
Both forms exist in the wild?

Reply:
They might, although in practice Windows doesn't seem to run on any big-endian systems, unless you count some vintages of XBox. I'm inclined to leave it in because it's cheap to adapt anyway, but could be persuaded otherwise.